src/Processors/QueryPlan/ReadFromMergeTree.cpp

```diff
@@ -978,18 +981,16 @@ MergeTreeDataSelectAnalysisResultPtr ReadFromMergeTree::selectRangesToRead(
     return std::make_shared<MergeTreeDataSelectAnalysisResult>(MergeTreeDataSelectAnalysisResult{.result = std::move(result)});
 }
 
-ReadFromMergeTree::AnalysisResult ReadFromMergeTree::getAnalysisResult() const
+void ReadFromMergeTree::checkAnalysisResult() const
```

Comment:
Let's keep the `getAnalysisResult()` but returning reference instead. We can build two variants:

```
    AnalysisResult & getAnalysisResult();
    const AnalysisResult & getAnalysisResult() const
    {
        return const_cast<ReadFromMergeTree *>(this)->getAnalysisResult();
    }
```